.github/workflows/release.yml

```diff
@@ -0,0 +1,129 @@
+
+name: Build release
+
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'Version to build'
+        required: false
+        default: ''
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    steps:
+    - name: "Check if user is in group"
+      run: |
+          MEMBER=$( curl -s \
+                 --request GET \
+                 --header 'authorization: Bearer ${{ secrets.GITTOKEN }}' \
+                 --header 'content-type: application/json' \
+                 --url https://api.github.com/orgs/matomo-org/teams/release-team/members \
+                 | jq '.[] | select(.login == "${{ github.actor }}")' )
+          
+          if [[ -z "$MEMBER" ]]
+          then
+           echo "Action was not triggered by a member with release permission"
+           exit 1;
+          fi
+      shell: bash
+    - uses: actions/checkout@v2
+      with:
+        lfs: false
+    - name: Import GPG key
+      id: import_gpg
+      uses: crazy-max/ghaction-import-gpg@v3
```

Comment:
same here let's point to a specific commit SHA.

Or can we replace this with a regular command so we don't need to use this action?

The `with` does this mean their script gets our private key and passphrase?

Reply:
Correct. Their script gets the private key and passphrase in order to set up GPG in a way so our script can use the key without the need to enter the password.

I'm pretty sure we could replace that with bash commands. But I don't know how to achieve that. Tried that when setting up this action for a while, but failed and then decided to look for an action that is able to achieve that.

If someone is keen to replace that with proper bash commands, feel free to give that a try.

Reply:
Looking at https://github.com/crazy-max/ghaction-import-gpg/blob/master/src/gpg.ts#L112 it executes `gpg --import --batch --yes $keyFile`?  (saw something similar in https://superuser.com/questions/1135812/gpg2-asking-for-passphrase-when-importing-secret-keys )

re the passphrase. It seems to be setting up an agent in https://github.com/crazy-max/ghaction-import-gpg/blob/master/src/main.ts#L41-L50 ? by executing https://github.com/crazy-max/ghaction-import-gpg/blob/master/src/gpg.ts#L35 eg if I see this right something along those lines:

```
echo "default-cache-ttl 7200
max-cache-ttl 31536000
allow-preset-passphrase" > $HOME/.gnupg/gpg-agent.conf
gpg-connect-agent "RELOADAGENT" /bye
gpg-connect-agent "PRESET_PASSPHRASE ${keygrip} -1 ${hexPassphrase}" /bye
gpg-connect-agent "KEYINFO ${keygrip}" /bye
```

Maybe this helps to find otherwise some further instructions/help for this?

Reply:
seeing also eg https://stackoverflow.com/questions/49491679/how-to-enter-gnupg-agent-key-passhprase-from-cli it seems important to allow preseting the passphrase and then use above commands to preset it

Reply:
can give that a try if you think forcing the action to a specific commit isn't enough. But might take a while to figure out the exact commands to get that working fully

Reply:
I suppose the alternative would be doing a review of the code to make sure there's nothing suspicious done. I suppose technically someone could [brute force a specific commit hash](https://github.com/bradfitz/gitbrute) but it's unlikely. Could maybe time box to 60 minutes to make it work locally since above commands should pretty much do what they should and otherwise do a quick review of the other action and use a specific hash 👍